src/serialization.rs

```diff
@@ -0,0 +1,61 @@
+use std::error::Error;
+
+use rustc_serialize::base64::{FromBase64, STANDARD, ToBase64};
+use serde::{Deserializer, Error as SerdeError, Serializer};
+use serde::de::Visitor;
+
+pub trait SerdeBlob: Sized {
+    fn deserialize_blob<D>(deserializer: &mut D) -> Result<Self, D::Error> where D: Deserializer;
+
+    fn serialize_blob<S>(&self, serializer: &mut S) -> Result<(), S::Error> where S: Serializer;
+}
+
+struct BlobVisitor;
+
+impl Visitor for BlobVisitor {
+    type Value = Vec<u8>;
+
+    fn visit_str<E>(&mut self, v: &str) -> Result<Self::Value, E> where E: SerdeError {
+        v.from_base64().map_err(|err| SerdeError::custom(err.description()))
```

Comment:
Sooooo this line is running `from_base64()` then `map_err` on the result, which will return the result if it works.  If not, it will execute the closure `SerdeError::custom(err.description())` with the argument of `err` which is the error from `from_base64()`?